### ISSUE TITLE
useScroll not working in Nuxt 3 with html node assigned using vue ref()

### ISSUE THREAD

**Issue by toddpadwick** [vueuse/vueuse#2771]
### Describe the bug

I am using `"@vueuse/core": "^9.12.0"` and `"@vueuse/nuxt": "^9.12.0"`, with nuxt 3. I have the following set up but useScroll does not seem to be updating the scroll value.

In other areas of my site, I have used vuescroll with window rather than refs, by assigning it on the mounted hook. This works, which indicates the issue is with HTML nodes and refs.

```
<template>
  <nav class="megamenu">
    <section class="megamenu__section">
    <div class="megamenu__section__content-slider__overflow" ref="productOverflow" :style="{'--scrollX':x ? x+'px' : '0px'}">
      // content
    </div>
  </section>
  </nav>
</template>


<script setup>
  import {ref, toRefs} from 'vue';
  import { useMediaQuery, useScroll } from '@vueuse/core';

  const productOverflow = ref(null);
  const { x } = useScroll(productOverflow.value);


</script>


```

I have also tried the following based on variations I saw in the docs:

```
  const productOverflow = ref<HTMLElement | null>(null); 
  const { x } = useScroll(productOverflow.value);
```

```
  const productOverflow = ref<HTMLElement | null>(null);
  const { x } = useScroll(productOverflow); / without .value
```

```
  const productOverflow = ref(null);
  const { x } = useScroll(productOverflow); // without .value
```

### Reproduction

Havn't had a chance to do this yet as I have a tight deadline - hoping this is a documentation problem

### System Info

```Shell
System:
    OS: macOS 11.7.3
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 65.64 MB / 32.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 14.19.0 - ~/.nvm/versions/node/v14.19.0/bin/node
    npm: 6.14.16 - ~/.nvm/versions/node/v14.19.0/bin/npm
  Browsers:
    Chrome: 109.0.5414.119
    Firefox: 105.0.1
    Safari: 16.3
    Safari Technology Preview: 15.4
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by okxiaoliang4** [comment]
What's `vuescroll/nuxt`? i did't find the module, looks fine for me. https://stackblitz.com/edit/nuxt-starter-vnzo85?file=app.vue

**Comment by toddpadwick** [comment]
Ah sorry, `vuescroll/nuxt` was a mistake in the ticket write up. I've corrected it now. 
I figured out what the problem was. My menu component hides the menu section with v-if, until one of the tabs is mouseover. which means, until that mouseover event happens, the `productOverflow` does not exist in the DOM and therefore the ref is null breaking the useScroll function. So what i have done, is pulled out the productOverflow element into its own component, that contains the useScroll functionality so the useScroll function is only triggered when its revealed.
Would be great if the useScroll function could listen for the ref change and initiate when its populated though, as in other cases it may not be feasible to pull the element out into its own component.
